python.d/ntp.chart.py

```diff
@@ -0,0 +1,434 @@
+        ]},
+    'sys_stratum': {
+        'options': [None, "Stratum (1-15)", "1", 'system', 'ntp.sys_stratum', 'line'],
+        'lines': [
+            ['stratum', 'stratum', 'absolute', 1, PRECISION]
+        ]},
+    'sys_tc': {
+        'options': [None, "Time constant and poll exponent (3-17)", "log2 s", 'system', 'ntp.sys_tc', 'line'],
+        'lines': [
+            ['tc', 'current', 'absolute', 1, PRECISION],
+            ['mintc', 'minimum', 'absolute', 1, PRECISION]
+        ]},
+    'sys_precision': {
+        'options': [None, "Precision", "log2 s", 'system', 'ntp.sys_precision', 'line'],
+        'lines': [
+            ['precision', 'precision', 'absolute', 1, PRECISION]
+        ]}
+}
+
+# Dynamic charts templates
+PEER_PREFIX = 'peer'
+
+PEER_DIMENSIONS = [
+    ['offset', 'Filter offset', 'ms'],
+    ['delay', 'Filter delay', 'ms'],
+    ['dispersion', 'Filter dispersion', 'ms'],
+    ['jitter', 'Filter jitter', 'ms'],
+    ['xleave', 'Interleave delay', 'ms'],
+    ['rootdelay', 'Total roundtrip delay to the primary reference clock', 'ms'],
+    ['rootdisp', 'Total root dispersion to the primary reference clock', 'ms'],
+    ['stratum', 'Stratum (1-15)', '1'],
+    ['hmode', 'Host mode (1-6)', '1'],
+    ['pmode', 'Peer mode (1-5)', '1'],
+    ['hpoll', 'Host poll exponent', 'log2 s'],
+    ['ppoll', 'Peer poll exponent', 'log2 s'],
+    ['precision', 'Precision', 'log2 s']
+]
+
+
+class Peer(object):
+    """
+    Class to hold peer data required in _get_data
+    """
+    def __init__(self, peer_id, name, request):
+        self.id = peer_id
+        self.name = name
+        self.request = request
+
+
+class Service(SocketService):
+    def __init__(self, configuration=None, name=None):
+        SocketService.__init__(self, configuration=configuration, name=name)
+        self.port = 'ntp'
+        self.dgram_socket = True
+        self.request_systemvars = None
+        self.peers = None
+        self.peer_error = 0
+        self.regex_srcadr = re.compile(r'srcadr=([A-Za-z0-9.-]+)')
+        self.regex_data = re.compile(r'([a-z_]+)=([0-9-]+(?:\.[0-9]+)?)(?=,)')
+        self.order = None
+        self.definitions = None
+
+        try:
+            peer_filter = r'^((0\.0\.0\.0)|(' + str(self.configuration['peer_filter']) + r'))$'
+        except (KeyError, TypeError):
+            self.error('error parsing peer_filter')
+            peer_filter = r'^((0\.0\.0\.0)|(127\..*))$'
+        self.regex_peer_filter = re.compile(peer_filter)
+
+        try:
+            self.peer_names = bool((self.configuration['peer_names']))
+        except (KeyError, TypeError):
+            self.peer_names = True
+
+    def create_charts(self):
+        """
+        Creates the charts dynamically.
+        Checks ntp for available peers.
+        Adds all peers whith valid data.
+        """
+        # Create systemvars charts
+        self.order = ORDER
+        self.definitions = CHARTS
```

Comment:
```python
self.order = list(ORDER)
self.definitions = dict(CHARTS)
```

Reply:
Awesome, that fixed the job problem right? I would never have seen that..